src/services/services.ts

```diff
@@ -2928,8 +2928,31 @@ namespace ts {
             log("getCompletionData: Is inside comment: " + (new Date().getTime() - start));
 
             if (insideComment) {
-                log("Returning an empty list because completion was inside a comment.");
-                return undefined;
+                // Completion should work inside certain JsDoc tags. For example:
+                //     /** @type {number | string} */
+                // Completion should work in the brackets
+                let insideJsDocTagExpression = false;
+                let tag = getJsDocTagAtPosition(sourceFile, position);
+                if (tag) {
+                    switch (tag.kind) {
+                        case SyntaxKind.JSDocTypeTag:
+                            let typeTag = <JSDocTypeTag>tag;
+                            if (typeTag.typeExpression) {
+                                insideJsDocTagExpression = position > typeTag.typeExpression.pos && position < typeTag.typeExpression.end;
+                            };
+                            break;
+                        case SyntaxKind.JSDocParameterTag:
+                            let paramTag = <JSDocParameterTag>tag;
+                            if (paramTag.typeExpression) {
+                                insideJsDocTagExpression = position > paramTag.typeExpression.pos && position < paramTag.typeExpression.end;
+                            };
+                            break;

```

Comment:
These branches are identical, just let each fall through and use a variable `typeTagOrParamTag` of type `JSDocTypeTag | JSDocParamTag`.